GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1285,13 +1265,54 @@ void OnRevisionReadCompleted()
                 // This has to happen on the UI thread
                 ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
```

Comment:
This comment is valid.
`await this.SwitchToMainThreadAsync();` must be done before both `AddArtificialRevisions` and `_gridView.SetToBeSelectedFromParents` at least because they are accessing the `_gridView` UI.

Reply:
Removing this comment, not valid in general.
_gridView.Add() is called from worker thread in OnRevisionRead(), same here.